101-data-factory-chained-copy-activities/azuredeploy.json

```diff
@@ -0,0 +1,370 @@
+        }
+      },
+      "azureSQLConnectionString": {
+        "type": "securestring",
+        "metadata": {
+          "description": "Connection string for Azure SQL Database"
+        }
+      }
+    },
+    "variables": {
+      "dataFactoryName": "[concat('CopyFromSQLToADFToAzureSQL', uniqueString(resourceGroup().id))]",
+      "gatewayName": "[concat('Gateway', uniqueString(resourceGroup().id))]",
+      "dataLakeLinkedServiceName": "DataFactoryLinkedService",
+      "sqlLinkedServiceName": "SqlLinkedService",
+      "azureSqlLinkedServiceName": "AzureSqlLinkedService",
+      "dataLakeDatasetName": "DataLakeDataset",
+      "sqlDatasetName": "SqlDataset",
+      "azureSqlDatasetName": "AzureSqlDataset",
+      "pipelineName": "CopyFromSQLToADFToAzureSQLPipeline"
+    },
+    "resources": [
+      {
+        "name": "[variables('dataFactoryName')]",
+        "apiVersion": "2015-10-01",
+        "type": "Microsoft.DataFactory/datafactories",
+        "location": "[resourceGroup().location]",
+        "resources": [
+            {
+                "dependsOn": [ "[concat('Microsoft.DataFactory/dataFactories/', variables('dataFactoryName'))]" ],
+                "type": "gateways",
+                "apiVersion": "2015-10-01",
+                "name": "[variables('gatewayName')]",
+                "properties": {
+                    "description": "ADF on-premises Data Management Gateway"
+                }
+            },    
+          {
+            "type": "linkedservices",
+            "name": "[variables('dataLakeLinkedServiceName')]",
+            "apiVersion": "2015-10-01",
+            "dependsOn": [
+              "[variables('dataFactoryName')]"
+            ],
+            "properties": {
+              "type": "AzureDataLakeStore",
+              "typeProperties": {
+                  "dataLakeStoreUri": "[parameters('dataLakeStoreUri')]",
+                  "servicePrincipalId": "[parameters('dataLakeStoreServicePrincipalID')]",
+                  "servicePrincipalKey": "[parameters('dataLakeStoreServicePrincipalKey')]",
+                  "tenant": "[subscription().tenantId]",
+                  "subscriptionId": "[subscription().subscriptionId]",
+                  "resourceGroupName": "[resourceGroup().name]"
+              }
+            }
+          },
+          {
+            "type": "linkedservices",
+            "name": "[variables('sqlLinkedServiceName')]",
+            "apiVersion": "2015-10-01",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('gatewayName')]"
+            ],
+            "properties": {
+              "type": "OnPremisesSqlServer",
+              "typeProperties": {
+                  "connectionString": "Placeholder-COnnectionString-Replace-Post-Deployment",
+                  "gatewayName": "[variables('gatewayName')]",
+                  "userName": "",
```

Comment:
are these properties required?  they look to have bogus values and there is a connstr...

Reply:
The property is required in order for the resource creation to succeed on deploy. There are explicit instructions in the README.md however to replace this value with an encrypted connection string post deployment (lines 38,39). The string cannot be encrypted until the deployment succeeds.